src/parser/smt2/Smt2.g

```diff
@@ -1682,13 +1675,9 @@ identifier[cvc5::ParseOp& p]
           p.d_kind = k;
           p.d_expr = SOLVER->mkInteger(numerals[0]);
         }
-        else if (numerals.size() == 1)
-        {
-          p.d_op = SOLVER->mkOp(k, numerals[0]);
-        }
-        else if (numerals.size() == 2)
+        else if (numerals.size() == 1 || numerals.size() == 2)
```

Comment:
I find this a bit weird. It seems to do a sanity check that the number of indices matches, but only goes half the way. Also, `mkOp()` does the proper check. For an operator that expects two indices, specifying only 1 or 3 gives very different errors.
How about just calling `mkOp` unconditionally, catch the exception (if there is one) and then passing the exception on via `parseError`?

Reply:
Hmm, I'm not sure if we should change this behavior in this PR.

Reply:
Fair enough. It's neither difficult nor particularly important to change later on, I guess...